src/Compiler/Checking/CheckExpressions.fs

```diff
@@ -6100,10 +6100,12 @@ and RewriteRangeExpr synExpr =
     match synExpr with
     // a..b..c (parsed as (a..b)..c )
     | SynExpr.IndexRange(Some (SynExpr.IndexRange(Some synExpr1, _, Some synStepExpr, _, _, _)), _, Some synExpr2, _m1, _m2, mWhole) ->
+        let mWhole = mWhole.MakeSynthetic()
```

Comment:
This range probably doesn't need to be made synthetic does it?  I mean, it should be accurate shouldn't it?

Reply:
I found that some/many other cases use synthetic ranges when rewriting expressions, and it's a good way to distinguish the user code from the compiler-generated, which we need in tooling.

Another option here is to produce two distinct non-synthetic ranges that would only cover the `..`. @dsyme What do you think?